### PR TITLE
Fix some things for pandas 3

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl!

File: dask_expr/_indexing.py [hunk 1]
```diff
@@ -133,13 +135,13 @@ def _loc_array(self, iindexer, cindexer):
...
 
-    def _maybe_partial_time_string(self, iindexer):
+    def _maybe_partial_time_string(self, iindexer, unit):
         """
         Convert index-indexer for partial time string slicing
         if obj.index is DatetimeIndex / PeriodIndex
         """
         idx = meta_nonempty(self.obj._meta.index)
-        iindexer = _maybe_partial_time_string(idx, iindexer)
+        iindexer = _maybe_partial_time_string(idx, iindexer, unit)
```
Comment (QuLogic) [inline dask_expr/_indexing.py:144]: This new argument to a private function means you need to depend on `dask>=2024.8.0` now.

Comment (jrbourbeau) [reply dask_expr/_indexing.py:144]: We updated our pinning to `dask == 2024.8.0` in the release earlier today https://github.com/dask/dask-expr/commit/837d44c56c97d463aa12cdcae47c626a458413f8

Comment (QuLogic) [reply dask_expr/_indexing.py:144]: Ack, I'm not sure why this didn't come through on our end. I'll investigate, but it's probably something we messed up somewhere.

Comment (jrbourbeau) [reply dask_expr/_indexing.py:144]: No worries -- let me know if changes are needed here

**Pull request merged**